Resources/doc/reference/configuration/proxy-client.rst

```diff
@@ -400,6 +400,15 @@ example to send a basic authentication header with each request, you can
 configure a service for the ``HttpClient`` and specify that in the
 ``http_client`` option of any of the cache proxy clients.
 
+.. _custom HTTP request factory:
+
+Custom HTTP Request Factory
+---------------------------
+
+The proxy client uses an implementation of ``Http\Message\RequestFactory`` to create HTTP requests.
```

Comment:
I think this is different for `FOSHttpCache` `v3` (it changed to `Psr\Http\Message\RequestFactoryInterface`) but `v3` of the bundle supports `v2` and `v3` of the lib. How do we properly document that?

Reply:
See `v2`: https://github.com/FriendsOfSymfony/FOSHttpCache/blob/2.x/src/ProxyClient/HttpProxyClient.php#L59
vs `v3`: https://github.com/FriendsOfSymfony/FOSHttpCache/blob/3.x/src/ProxyClient/HttpProxyClient.php#L51

Reply:
good point. i actually don't see value in keeping to support FOSHttpCache 2.x with the 3.x bundle. both have the same minimum PHP version, so there is no good argument to use different major versions.

i will adjust accordingly.

Reply:
ah, and we also added a psr-17 stream factory optional argument. for consistency we should support that as well.

Reply:
Should I create a PR for that?

Reply:
thanks. i already committed it in this branch :-D